src/handlers/synthetic-events/business-logic.ts

```diff
@@ -13,29 +38,82 @@ export const businessLogic: BusinessLogic<
 > = async (_, { config }) => {
   const syntheticEventsConfig = config[ConfigElements.syntheticEvents];
 
+  const dashboardData = await getDashboardExtract();
+
   const now = new Date();
-  const nowTime = now.getTime();
-  const nowFormatted = formatDate(now);
   const events: CleanedEventBody[] = [];
 
-  syntheticEventsConfig.forEach((configLine) => {
-    if (
-      new Date(configLine.start_date).getTime() < nowTime &&
-      (!configLine.end_date ||
-        new Date(configLine.end_date).getTime() > nowTime) &&
-      configLine.frequency === "monthly"
-    ) {
-      const event: CleanedEventBody = {
-        vendor_id: configLine.vendor_id,
-        event_id: crypto.randomUUID(),
-        event_name: configLine.event_name,
-        timestamp: nowTime,
-        timestamp_formatted: nowFormatted,
-        credits: configLine.quantity,
-        component_id: configLine.component_id,
-      };
-      events.push(event);
-    }
+  syntheticEventsConfig.forEach((syntheticEventDefinition) => {
```

Comment:
This could also be a chubby reduce, doesn't save you any loops but you don't have to declare `events` and push into it 

```
  return syntheticEventsConfig.reduce<CleanedEventBody[]>(
    (acci, syntheticEventDefinition) => {
      const activePeriods = getActivePeriods(
        now,
        syntheticEventDefinition.frequency,
        syntheticEventDefinition.type,
        new Date(syntheticEventDefinition.start_date),
        syntheticEventDefinition.end_date
          ? new Date(syntheticEventDefinition.end_date)
          : undefined
      );

      return [
        ...acci,
        ...activePeriods.reduce<CleanedEventBody[]>((accj, period) => {
          const existingEventCount = countExistingEvents(
            syntheticEventDefinition,
            period,
            dashboardData
          );

          if (syntheticEventDefinition.quantity < existingEventCount)
            return accj;

          return [
            ...accj,
            createEvent(
              new Date(period.year, period.month - 1),
              syntheticEventDefinition,
              syntheticEventDefinition.quantity - existingEventCount
            ),
          ];
        }, []),
      ];
    },
    []
  );
```